<!DOCTYPE html><html lang="zh-CN"><head><meta http-equiv="content-type" content="text/html; charset=utf-8"><meta content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=0" name="viewport"><meta content="yes" name="apple-mobile-web-app-capable"><meta content="black-translucent" name="apple-mobile-web-app-status-bar-style"><meta content="telephone=no" name="format-detection"><meta name="description"><title>Windows系统下16S rDNA扩增子数据分析流程 | LIGENE</title><link rel="stylesheet" type="text/css" href="/css/style.css?v=1.0.0"><link rel="stylesheet" type="text/css" href="//lib.baomitu.com/normalize/latest/normalize.min.css"><link rel="stylesheet" type="text/css" href="//lib.baomitu.com/pure/latest/pure-min.min.css"><link rel="stylesheet" type="text/css" href="//lib.baomitu.com/pure/latest/grids-responsive-min.min.css"><link rel="stylesheet" href="//lib.baomitu.com/font-awesome/4.7.0/css/font-awesome.min.css"><script type="text/javascript" src="//lib.baomitu.com/jquery/latest/jquery.min.js"></script><link rel="icon" mask="" sizes="any" href="/favicon.ico"><link rel="Shortcut Icon" type="image/x-icon" href="/favicon.ico"><link rel="apple-touch-icon" href="/apple-touch-icon.png"><link rel="apple-touch-icon-precomposed" href="/apple-touch-icon.png"><link rel="alternate" type="application/atom+xml" href="/atom.xml"><script>var _hmt = _hmt || [];
(function() {
  var hm = document.createElement('script');
  hm.src = 'https://hm.baidu.com/hm.js?' + '0ad9c65dc343a2a6d37cb95527713113';
  var s = document.getElementsByTagName("script")[0];
  s.parentNode.insertBefore(hm, s);
  })();</script><script type="text/javascript" src="//lib.baomitu.com/clipboard.js/latest/clipboard.min.js"></script><script type="text/javascript" src="//lib.baomitu.com/toastr.js/latest/toastr.min.js"></script><link rel="stylesheet" href="//lib.baomitu.com/toastr.js/latest/toastr.min.css"><meta name="generator" content="Hexo 6.0.0"></head><body><div class="body_container"><div id="header"><div class="site-name"><h1 class="hidden">Windows系统下16S rDNA扩增子数据分析流程</h1><a id="logo" href="/.">LIGENE</a><p class="description">Notes of Bioinformatics and Genomics</p></div><div id="nav-menu"><a class="current" href="/."><i class="fa fa-home"> 首页</i></a><a href="/archives/"><i class="fa fa-archive"> 归档</i></a><a href="/../bigbook/"><i class="fa fa-book"> 教程</i></a><a href="/downloads/"><i class="fa fa-download"> 下载</i></a><a href="/about/"><i class="fa fa-user"> 关于</i></a></div></div><div class="pure-g" id="layout"><div class="pure-u-1 pure-u-md-3-4"><div class="content_container"><div class="post"><h1 class="post-title">Windows系统下16S rDNA扩增子数据分析流程</h1><div class="post-meta">2019-06-08<span> | </span><span class="category"><a href="/categories/Genomics/">Genomics</a></span><span class="post-time"><span class="post-meta-item-text"> | </span><span class="post-meta-item-icon"><i class="fa fa-keyboard-o"></i><span class="post-count"> 2.7k</span><span class="post-meta-item-text"> 字</span></span></span><span class="post-time"> | <span class="post-meta-item-icon"><i class="fa fa-clock-o"></i><span class="post-count"> 12</span><span class="post-meta-item-text"> 分钟</span></span></span></div><a class="disqus-comment-count" href="/2019/06/08/win16s-pipeline/#vcomment"><span class="valine-comment-count" data-xid="/2019/06/08/win16s-pipeline/"></span><span> 条评论</span></a><div class="post-content"><p>本16S rRNA基因扩增子测序数据分析流程包括：首先对样本测序数据进行质控与预处理；然后得到样本的OTU聚类与物种分类信息；在查看样品的细菌种类后，统计每个样品的差异菌种。For more details about 16S rRNA analysis, recommend following the tutorial: <a target="_blank" rel="noopener" href="https://rachaellappan.github.io/16S-analysis/index.html">https://rachaellappan.github.io/16S-analysis/index.html</a><br>Please note that QIIME1 and the 97% OTU-based workflow has been superseded by ASVs (100% OTUs) and the QIIME2 workflow! </p>
<span id="more"></span>

<h3 id="1-流程相关软件安装"><a href="#1-流程相关软件安装" class="headerlink" title="1. 流程相关软件安装"></a>1. 流程相关软件安装</h3><p>本分析流程需要在Windows的Linux子系统(WSL)下安装QIIME、VSEARCH等软件。</p>
<ul>
<li><p>安装Linux常用工具</p>
<figure class="highlight plaintext"><table><tr><td class="code"><pre><span class="line">$sudo apt-get install build-essential	#Linux基础开发工具</span><br><span class="line">$sudo apt-get install python-minimal python-dev python-tk </span><br><span class="line">$sudo apt-get install python-pip  #可用pip安装python包</span><br><span class="line">$python --version</span><br><span class="line">#如果python不是python2.7，请设置python2为默认版本：alias python=’/usr/bin/python2’(可把此命令写入~/.bashrc文件中)。</span><br></pre></td></tr></table></figure>
</li>
<li><p>安装QIIME</p>
<figure class="highlight plaintext"><table><tr><td class="code"><pre><span class="line">$sudo pip install --upgrade pip -i http://mirrors.aliyun.com/pypi/simple --trusted-host mirrors.aliyun.com</span><br><span class="line">$sudo pip install numpy==1.10.0 -i http://mirrors.aliyun.com/pypi/simple --trusted-host mirrors.aliyun.com</span><br><span class="line">$sudo pip install matplotlib==1.5.3 -i http://mirrors.aliyun.com/pypi/simple --trusted-host mirrors.aliyun.com</span><br><span class="line">$sudo pip install scipy==1.0.0 -i http://mirrors.aliyun.com/pypi/simple --trusted-host mirrors.aliyun.com</span><br><span class="line">$sudo pip install ipython==5.5.0 -i http://mirrors.aliyun.com/pypi/simple --trusted-host mirrors.aliyun.com</span><br><span class="line">sudo pip install pandas==0.21.1 -i http://mirrors.aliyun.com/pypi/simple --trusted-host mirrors.aliyun.com</span><br><span class="line">$sudo pip install qiime -i http://mirrors.aliyun.com/pypi/simple --trusted-host mirrors.aliyun.com</span><br><span class="line"></span><br><span class="line">#测序QIIME安装是否成功</span><br><span class="line">$print_qiime_config.py -t</span><br></pre></td></tr></table></figure>
</li>
<li><p>安装VSEARCH</p>
<figure class="highlight plaintext"><table><tr><td class="code"><pre><span class="line">$git clone https://github.com/torognes/vsearch.git</span><br><span class="line">$cd vsearch</span><br><span class="line">$./autogen.sh</span><br><span class="line">$./configure</span><br><span class="line">$make</span><br><span class="line">$sudo make install  #默认安装在/usr/local/bin/目录</span><br><span class="line">$sudo cp /usr/local/bin/vsearch /usr/local/bin/usearch61 #让qiime的参数usearch61使用vsearch程序</span><br></pre></td></tr></table></figure>
</li>
<li><p>安装ClustalW<br><code>$sudo apt-get install clustalw</code></p>
</li>
<li><p>安装RDP classifier</p>
<figure class="highlight plaintext"><table><tr><td class="code"><pre><span class="line">#1.下载RDP Classifier v2.2(只有此版本能用)，下载网址：https://sourceforge.net/projects/rdp-classifier/files/rdp-classifier/</span><br><span class="line">#2.解压到windows系统的D盘根目录，并复制到Ubuntu系统的用户home目录：</span><br><span class="line">$mv /mnt/d/rdp_classifier_2.2/  ~/</span><br><span class="line">$echo &quot;export RDP_JAR_PATH=~/rdp_classifier_2.2/rdp_classifier-2.2.jar&quot; &gt;&gt; ~/.bashrc</span><br><span class="line">$source ~/.bashrc</span><br></pre></td></tr></table></figure>
</li>
<li><p>安装Xming可运行X-windows图形程序</p>
<figure class="highlight plaintext"><table><tr><td class="code"><pre><span class="line">#下载安装包Xming-setup.exe（最新版本6.9.0.31），Xming官网：http://sourceforge.net/projects/xming/</span><br><span class="line">$echo &quot;export DISPLAY=:0.0&quot; &gt;&gt; ~/.bashrc #设置X图形显示端口</span><br><span class="line">$source ~/.bashrc</span><br></pre></td></tr></table></figure></li>
</ul>
<h3 id="2-流程运行命令"><a href="#2-流程运行命令" class="headerlink" title="2. 流程运行命令"></a>2. 流程运行命令</h3><p>此流程所有命令都需在WSL-Ubuntu的bash终端下运行（以下命令前面的“$”代表bash提示符，不需要输入）。运行Ubuntu之前，先在电脑C盘创建目录win16s，并将样本测序数据文件(raw_reads.zip)，mapping.txt，参考序列文件(rdp_gold.fa)与参数设置文件(qiime_params.txt)复制到此目录下。<br>注意：所有输入文件名及目录，根据自己保存的文件进行调整。</p>
<h4 id="1-准备测序数据-raw-reads"><a href="#1-准备测序数据-raw-reads" class="headerlink" title="(1)准备测序数据(raw reads)"></a>(1)准备测序数据(raw reads)</h4><p>本实验需要把所有原始数据放在一个文件夹，一个样品的双端测序有两个reads文件，默认以”_R1”与”_R2”分别代表正向与反向测序reads。测序公司提供的数据一般已去除barcode和primer序列。</p>
<figure class="highlight plaintext"><table><tr><td class="code"><pre><span class="line">$cd /mnt/c/ch17rRNA/		#切换到工作目录</span><br><span class="line">$unzip raw_reads.zip</span><br><span class="line">$ls raw_reads</span><br><span class="line">显示文件列表，如SAA_R1.fastq.gz, SAA_R2.fastq.gz等，如果文件名不对，需要重新命名。</span><br></pre></td></tr></table></figure>
<h4 id="2-准备mapping-file"><a href="#2-准备mapping-file" class="headerlink" title="(2)准备mapping file"></a>(2)准备mapping file</h4><p>mapping文件必须的列名如下：<br><code>#sampleID    BarcodeSequence    LinkerPrimerSequence    Description</code><br>注意：每项之间不能出现空格, 间隔是以TAB键隔开。如果没有BarcodeSequence与LinkerPrimerSequennce信息可以为空。<br>LinkerPrimerSequence与Description之间可以有任意项来说明样本的额外信息。如要修改可用Excel打开mapping.txt，按样本名称修改，再保存为txt格式即可。<br>本测序数据样本mapping file式如下：</p>
<figure class="highlight plaintext"><table><tr><td class="code"><pre><span class="line">#SampleID	BarcodeSequence	LinkerPrimerSequence    PRIMER	Group	Description</span><br><span class="line">SAA	ACGAGTGCGTA	GAGTTTGATCMTGGCTC 8F	diet	SAA</span><br><span class="line">SAB	AGACGCACTCA	GAGTTTGATCMTGGCTC 8F	fat	SAB</span><br></pre></td></tr></table></figure>
<p>这里SampleID项对应于前面测序reads文件名的前三个字母， Group是样本的分组信息, PRIMER是16S引物名称。<br>修改mapping.txt后执行以下命令检查是否有问题：<br><code>$validate_mapping_file.py -m mapping.txt -o validate_map</code><br>在Wiondows资源管理器下打开输出目录validate_map中的mapping.html文件查看检查信息，并根据错误提示修改mapping文件。如果没有BarcodeSequence与LinkerPrimerSequennce信息，可忽略Barcode与Primer序列为空的错误提示。</p>
<h4 id="3-合并双端序列"><a href="#3-合并双端序列" class="headerlink" title="(3)合并双端序列"></a>(3)合并双端序列</h4><p>如果样品数量较小，可以对每个样品分别进行合并，一次合并一个样品的两端测序数据：</p>
<figure class="highlight plaintext"><table><tr><td class="code"><pre><span class="line">$mkdir  join_pe_reads</span><br><span class="line">$vsearch --fastq_mergepairs raw_reads/SAA_R1.fastq.gz --reverse raw_reads/SAA_R2.fastq.gz --fastqout join_pe_reads/SAA.fq</span><br><span class="line">Merging reads 100%</span><br><span class="line">     28741  Pairs</span><br><span class="line">     19633  Merged (68.3%)</span><br><span class="line">      9108  Not merged (31.7%)</span><br></pre></td></tr></table></figure>
<p>如果样品数量较多，也可用以下shell脚本一次合并多个样品数据：</p>
<figure class="highlight plaintext"><table><tr><td class="code"><pre><span class="line">$for i in `tail -n+2 mapping.txt |cut -f 1`; do vsearch --fastq_mergepairs raw_reads/$&#123;i&#125;_R1.fastq.gz --reverse raw_reads/$&#123;i&#125;_R2.fastq.gz --fastqout join_pe_reads/$&#123;i&#125;.fq ; done</span><br></pre></td></tr></table></figure>
<h4 id="4-序列质量控制"><a href="#4-序列质量控制" class="headerlink" title="(4)序列质量控制"></a>(4)序列质量控制</h4><p>一般可先用FastQC检查测序reads的质量信息，注意测序reads长度。16S rRNA v3-v4区片段大约为400bp左右，这里将合并后reads长度&lt; 380 bp的reads都过滤掉。</p>
<figure class="highlight plaintext"><table><tr><td class="code"><pre><span class="line">$mkdir qual_filter</span><br><span class="line">$vsearch --fastx_filter join_pe_reads/SAA.fq --fastq_maxee 1.0 --fastq_minlen 380 --fastqout join_pe_reads/SAA_filter.fq</span><br></pre></td></tr></table></figure>
<p>如果有多个样品数据，可直接copy下面命令到terminal执行：</p>
<figure class="highlight plaintext"><table><tr><td class="code"><pre><span class="line">$for i in `tail -n+2 mapping.txt |cut -f 1`; do vsearch --fastx_filter join_pe_reads/$&#123;i&#125;.fq --fastq_maxee 1.0 --fastq_minlen 380 --fastqout qual_filter/$&#123;i&#125;.fq ; done</span><br></pre></td></tr></table></figure>

<h4 id="5-割库-split-libraries"><a href="#5-割库-split-libraries" class="headerlink" title="(5)割库(split libraries)"></a>(5)割库(split libraries)</h4><p>此步骤将所有样本的序列数据合并成到一个文件，并以样品名区分样品序列。</p>
<figure class="highlight plaintext"><table><tr><td class="code"><pre><span class="line">$mkdir split_out</span><br><span class="line">$multiple_split_libraries_fastq.py -i qual_filter/ -o split_out/ --demultiplexing_method sampleid_by_file</span><br></pre></td></tr></table></figure>
<p>输出文件为 seqs.fna，每条序列的ID按来源的样品重命名，头行还包括纠正barcode错误(bc_diffs)等信息。<br>此命令运行的概要信息可以查看输出文件split_library_log.txt，包含样品的reads数量，低质量reads过滤结果等。</p>
<h4 id="6-Dereplication-序列去重复"><a href="#6-Dereplication-序列去重复" class="headerlink" title="(6)Dereplication(序列去重复)"></a>(6)Dereplication(序列去重复)</h4><p><code>$vsearch --derep_fulllength split_out/seqs.fna --output split_out/seqs_derep.fa --sizeout --minuniquesize 2</code></p>
<h4 id="7-嵌合体检测"><a href="#7-嵌合体检测" class="headerlink" title="(7)嵌合体检测"></a>(7)嵌合体检测</h4><p>嵌合子的产生主要是由于PCR过程中模版的不完全延伸。这一步是去除嵌合子，一般需要参考序列文件：<br><code>$vsearch --uchime_ref split_out/seqs_derep.fa --db rdp_gold.fa --nonchimeras seqs_nochimeras.fa </code><br>如果没有参考序列，VSEARCH也可以进行去嵌合体：<br><code>$vsearch --uchime_denovo split_out/seqs_derep.fa --nonchimeras seqs_nochimeras.fa</code></p>
<h4 id="8-OTU聚类-OTU-clustering"><a href="#8-OTU聚类-OTU-clustering" class="headerlink" title="(8)OTU聚类(OTU clustering)"></a>(8)OTU聚类(OTU clustering)</h4><p>OTU (operational taxonomic units)是人为给某一个分类单元(科、属、种等)设置的标志。通常按照97%的相似性阈值将序列划分为不同的OTU，每一个OTU通常被认为是一个微生物物种。选取最长的序列作为OTU的代表。<br><code>$vsearch --cluster_fast seqs_nochimeras.fa --id 0.97 --centroids otus.fa --relabel OTU_</code><br>以序列最小相似度97%聚类，得到Clusters: 121</p>
<h4 id="9-物种分类注释"><a href="#9-物种分类注释" class="headerlink" title="(9)物种分类注释"></a>(9)物种分类注释</h4><p>QIIME 采用RDP classifier 将OUT注释为不同物种分类。<br><code>$assign_taxonomy.py -i otus.fa -m rdp -o taxonomy</code><br>注意：分类数据库可以采用GreenGenes (<a target="_blank" rel="noopener" href="http://greengenes.secondgenome.com/)%E3%80%81Silva">http://greengenes.secondgenome.com/)、Silva</a> (<a target="_blank" rel="noopener" href="https://www.arb-silva.de/)%E5%92%8CRDP">https://www.arb-silva.de/)和RDP</a> (<a target="_blank" rel="noopener" href="http://rdp.cme.msu.edu/)%E3%80%82%E8%BF%99%E9%87%8C%E9%87%87%E7%94%A8Ribosomal">http://rdp.cme.msu.edu/)。这里采用Ribosomal</a> Database Project classifier与Greengenes 数据库（13_8版本）。 </p>
<h4 id="10-生成OTU表格"><a href="#10-生成OTU表格" class="headerlink" title="(10)生成OTU表格"></a>(10)生成OTU表格</h4><ul>
<li><ol>
<li>Map reads back to the OTU data：<br><code>$vsearch --usearch_global split_out/seqs.fna --db otus.fa --strand plus --id 0.97 --otutabout otus_table.txt</code></li>
</ol>
</li>
<li><ol start="2">
<li>Convert otu_table.txt to otu-table.biom<figure class="highlight plaintext"><table><tr><td class="code"><pre><span class="line">$biom convert -i otus_table.txt -o otus_table.biom --to-hdf5 --table-type &quot;OTU table&quot;</span><br><span class="line">$biom add-metadata -i otus_table.biom -o otus_table_tax.biom --observation-metadata-fp taxonomy/otus_tax_assignments.txt --sc-separated taxonomy --observation-header OTUID,taxonomy</span><br></pre></td></tr></table></figure></li>
</ol>
</li>
<li><p>也可以用QIIME的脚本生成OTU表：<br><code>make_otu_table.py -i otus_table.txt -t taxonomy/otus_tax_assignments.txt -o otus_table1.biom</code></p>
</li>
</ul>
<h4 id="11-检查OTU-Table"><a href="#11-检查OTU-Table" class="headerlink" title="(11)检查OTU Table"></a>(11)检查OTU Table</h4><p><code>$biom summarize-table -i otus_table_tax.biom -o summary_biom.txt</code><br>记录样本测序数据量最少的read Counts数(此处为Min：11640），用于后续的统计抽样（步骤15中的-e参数）。</p>
<ul>
<li>关于OTU表的更多检查问题参见后面注释(Notes)部分。</li>
</ul>
<p>16S数据一般有许多noise，如嵌合体序列，低丰度的OTUs等。<br>我们只过滤低频率的OTUs，过滤的标准是OTU总数在每个样品中都低于2(-min_count 2)，而且任何OTU只在少于25%的样品中有(-min_samples 3):</p>
<figure class="highlight plaintext"><table><tr><td class="code"><pre><span class="line">$filter_otus_from_otu_table.py -i otus_table_tax.biom -o otus_table_tax_filtered.biom  --min_count 2</span><br><span class="line">#查看还剩下多少OTU？</span><br><span class="line">$biom summarize-table -i otus_table_tax_filtered.biom</span><br><span class="line">//Num observations: 199</span><br></pre></td></tr></table></figure>

<p>注意如果发现有什么错误，可以先用Excel打开otu_table_tsv.txt修改，然后再转换成biom格式。</p>
<figure class="highlight plaintext"><table><tr><td class="code"><pre><span class="line">$biom convert -i otus_table_tax.biom -o otus_table_tsv.txt --table-type &quot;OTU table&quot; --to-tsv --header-key taxonomy</span><br><span class="line">$biom convert -i otus_table_tsv.txt  -o otus_table_tsv.biom --table-type &quot;OTU table&quot; --to-hdf5 --table-type=&quot;OTU table&quot; --process-obs-metadata taxonomy</span><br></pre></td></tr></table></figure>

<h4 id="12-可视化"><a href="#12-可视化" class="headerlink" title="(12)可视化"></a>(12)可视化</h4><p>运行可视化命令前，电脑要先运行Xming软件。</p>
<figure class="highlight plaintext"><table><tr><td class="code"><pre><span class="line">#Plot relative abundance for all samples</span><br><span class="line">$summarize_taxa_through_plots.py -i otus_table_tax.biom -m mapping.txt -o taxa_summary</span><br></pre></td></tr></table></figure>
<p>输出文件存放在目录taxa_summary，其中图形信息放在目录taxa_summary_plots，可用浏览器打开bar_charts.html文件查看各个样本自门水平(L2)到属水平(L6)分类水平的菌株组成分布信息。<br>从上到下分别列出5个分类水平门到属的柱状图，每个柱状图下面是对应的数据表，表中分类(Taxonomy)的字母分别代表不同分类水平：k&#x3D;kingdom, p&#x3D;phylum, c&#x3D;class, o&#x3D;order, f&#x3D;family, g&#x3D;genus.<br><img src="https://raw.githubusercontent.com/adong77/ligene/master/Images/book/taxa_plots.png" alt="taxa_bar_plots.png"><br>图 1 样品的菌群分类与相对丰度 (A)门水平； (B)属水平</p>
<h4 id="13-Align-sequences-on-QIIME-using-the-ClustalW-or-muscle-method"><a href="#13-Align-sequences-on-QIIME-using-the-ClustalW-or-muscle-method" class="headerlink" title="(13) Align sequences on QIIME using the ClustalW or muscle method"></a>(13) Align sequences on QIIME using the ClustalW or muscle method</h4><p><code>$align_seqs.py -i otus.fa -m clustalw -o rep_set_align</code></p>
<h4 id="14-Make-the-reference-tree-on-QIIME"><a href="#14-Make-the-reference-tree-on-QIIME" class="headerlink" title="(14) Make the reference tree on QIIME"></a>(14) Make the reference tree on QIIME</h4><p><code>$make_phylogeny.py -i rep_set_align/otus_aligned.fasta -o rep_set.tre</code></p>
<h4 id="15-多样性分析-Run-diversity-analyses-on-QIIME"><a href="#15-多样性分析-Run-diversity-analyses-on-QIIME" class="headerlink" title="(15)多样性分析(Run diversity analyses on QIIME)"></a>(15)多样性分析(Run diversity analyses on QIIME)</h4><p>多样性分析有Alpha diversity与Beta diversity。<br><code>$core_diversity_analyses.py -i otus_table_tax.biom -m mapping.txt -t rep_set.tre -e 10000 -o core_diversity -p qiime_params.txt</code><br>参数“-e 10000”是用于均匀抽样和最大稀疏(rarefaction)深度的测序深度。由于测序技术的原因，不同样品的数据量有差异，因此不同的样品数据要先随机从每个样品中选择相同数目的reads (rarefy the data)进行标准化处理后才能进行样品间的比较。</p>
<p>下面列举几个特殊的分析命令：</p>
<ul>
<li><ol>
<li>Making PCoA plots:<br><code>$make_2d_plots.py -i core_diversity/bdiv_even11640/weighted_unifrac_pc.txt -o pcoa_plot -m mapping.txt -b Group</code><br>注：-i参数后输入文件所在目录的数字（11640）会随前面命令的-e参数变化。</li>
</ol>
</li>
<li><ol start="2">
<li>统计两组间的菌群差异：<br><code>$compare_categories.py --method anosim -i core_diversity/bdiv_even11640/weighted_unifrac_dm.txt -m mapping.txt -c Group -o anosim_out -n 99</code></li>
</ol>
</li>
</ul>
<hr>
<h3 id="3-Notes："><a href="#3-Notes：" class="headerlink" title="3. Notes："></a>3. Notes：</h3><ul>
<li><ol>
<li>16S数据一般有许多noise，如嵌合体序列，低丰度的OTUs等。<br>下面的命令只过滤低频率的OTUs，过滤的标准是OTU总数在每个样品中都低于2(-min_count 2)，而且任何OTU只在少于25%的样品中有(-min_samples 3):<figure class="highlight plaintext"><table><tr><td class="code"><pre><span class="line">$filter_otus_from_otu_table.py -i otus_table_tax.biom -o otus_table_tax_filtered.biom  --min_count 2</span><br><span class="line">#查看还剩下多少OTU？</span><br><span class="line">$biom summarize-table -i otus_table_tax_filtered.biom</span><br><span class="line">//Num observations: 199</span><br></pre></td></tr></table></figure></li>
</ol>
</li>
<li><ol start="2">
<li>注意：如果发现有什么错误，可以先将biom文件转换成文本文件，再用Excel打开otu_table_tsv.txt修改，然后再转换成biom格式。<figure class="highlight plaintext"><table><tr><td class="code"><pre><span class="line">$biom convert -i otus_table_tax.biom -o otus_table_tsv.txt --table-type &quot;OTU table&quot; --to-tsv --header-key taxonomy</span><br><span class="line">$biom convert -i otus_table_tsv.txt  -o otus_table_tsv.biom --table-type &quot;OTU table&quot; --to-hdf5 --table-type=&quot;OTU table&quot; --process-obs-metadata taxonomy</span><br></pre></td></tr></table></figure></li>
</ol>
</li>
</ul>
<p><strong>For more details about 16S rRNA analysis</strong>, recommend following the tutorial: <a target="_blank" rel="noopener" href="https://rachaellappan.github.io/16S-analysis/index.html">https://rachaellappan.github.io/16S-analysis/index.html</a> </p>
</div><div class="post-copyright"><script type="text/javascript" src="/js/copyright.js?v=1.0.0" successtext="复制成功！"></script><link rel="stylesheet" type="text/css" href="/css/copyright.css?v=1.0.0"><p><span>本文标题：</span>Windows系统下16S rDNA扩增子数据分析流程</p><p><span>文章作者：</span>adong</p><p><span>发布时间：</span>2019-06-08</p><p><span>最后更新：</span>2023-01-17</p><p><span>原始链接：</span><a href="/2019/06/08/win16s-pipeline/">http://blog.ligene.cn/2019/06/08/win16s-pipeline/</a><span class="copy-path"><i class="fa fa-clipboard" data-clipboard-text="http://blog.ligene.cn/2019/06/08/win16s-pipeline/"></i></span></p><p><span>版权声明：</span>本站所有文章除特别声明外，均采用 CC BY-NC-SA 3.0 CN 许可协议。转载请注明出处！</p></div><br><div class="tags"><a href="/tags/16S"><i class="fa fa-tag">16S</i></a><a href="/tags/WSL"><i class="fa fa-tag">WSL</i></a></div><div class="post-nav"><a class="pre" href="/2019/06/09/SetupVPN/">科学上网简易方法</a><a class="next" href="/2018/01/25/Xming-PuTTY/">Windows下连接远程Linux桌面</a></div><div id="vcomment"></div><script src="//c.lcfile.com/static/js/av-core-mini-0.6.4.js"></script><script src="//unpkg.com/valine@latest/dist/Valine.min.js"></script><script>var notify = 'false' == 'true' ? true : false;
var verify = 'false' == 'true' ? true : false;
var GUEST_INFO = ['nick','mail','link'];
var guest_info = 'nick,mail'.split(',').filter(function(item){
  return GUEST_INFO.indexOf(item) > -1
});
guest_info = guest_info.length == 0 ? GUEST_INFO :guest_info;
window.valine = new Valine({
  el:'#vcomment',
  notify:notify,
  verify:verify,
  appId:'MNJKe5WJm5b2Ef1eGugrbjeW-9Nh9j0Va',
  appKey:'JidWqAC9XmARrOScdPmYsjOc',
  placeholder:'Writing comments here',
  avatar:'mm',
  guest_info:guest_info,
  pageSize:'10'
})
</script></div></div></div><script type="text/javascript" src="/js/toc.js?v=1.0.0" async></script><div class="pure-u-1-4 hidden_mid_and_down"><div id="sidebar-toc"><div class="stoc-article" id="sidebar-stoc"><strong class="stoc-title"><i class="fa fa-bars"> TOC </i></strong><div class="toc-nav" id="stoc"><ol class="toc"><li class="toc-item toc-level-3"><a class="toc-link" href="#1-%E6%B5%81%E7%A8%8B%E7%9B%B8%E5%85%B3%E8%BD%AF%E4%BB%B6%E5%AE%89%E8%A3%85"><span class="toc-number">1.</span> <span class="toc-text">1. 流程相关软件安装</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#2-%E6%B5%81%E7%A8%8B%E8%BF%90%E8%A1%8C%E5%91%BD%E4%BB%A4"><span class="toc-number">2.</span> <span class="toc-text">2. 流程运行命令</span></a><ol class="toc-child"><li class="toc-item toc-level-4"><a class="toc-link" href="#1-%E5%87%86%E5%A4%87%E6%B5%8B%E5%BA%8F%E6%95%B0%E6%8D%AE-raw-reads"><span class="toc-number">2.1.</span> <span class="toc-text">(1)准备测序数据(raw reads)</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#2-%E5%87%86%E5%A4%87mapping-file"><span class="toc-number">2.2.</span> <span class="toc-text">(2)准备mapping file</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#3-%E5%90%88%E5%B9%B6%E5%8F%8C%E7%AB%AF%E5%BA%8F%E5%88%97"><span class="toc-number">2.3.</span> <span class="toc-text">(3)合并双端序列</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#4-%E5%BA%8F%E5%88%97%E8%B4%A8%E9%87%8F%E6%8E%A7%E5%88%B6"><span class="toc-number">2.4.</span> <span class="toc-text">(4)序列质量控制</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#5-%E5%89%B2%E5%BA%93-split-libraries"><span class="toc-number">2.5.</span> <span class="toc-text">(5)割库(split libraries)</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#6-Dereplication-%E5%BA%8F%E5%88%97%E5%8E%BB%E9%87%8D%E5%A4%8D"><span class="toc-number">2.6.</span> <span class="toc-text">(6)Dereplication(序列去重复)</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#7-%E5%B5%8C%E5%90%88%E4%BD%93%E6%A3%80%E6%B5%8B"><span class="toc-number">2.7.</span> <span class="toc-text">(7)嵌合体检测</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#8-OTU%E8%81%9A%E7%B1%BB-OTU-clustering"><span class="toc-number">2.8.</span> <span class="toc-text">(8)OTU聚类(OTU clustering)</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#9-%E7%89%A9%E7%A7%8D%E5%88%86%E7%B1%BB%E6%B3%A8%E9%87%8A"><span class="toc-number">2.9.</span> <span class="toc-text">(9)物种分类注释</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#10-%E7%94%9F%E6%88%90OTU%E8%A1%A8%E6%A0%BC"><span class="toc-number">2.10.</span> <span class="toc-text">(10)生成OTU表格</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#11-%E6%A3%80%E6%9F%A5OTU-Table"><span class="toc-number">2.11.</span> <span class="toc-text">(11)检查OTU Table</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#12-%E5%8F%AF%E8%A7%86%E5%8C%96"><span class="toc-number">2.12.</span> <span class="toc-text">(12)可视化</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#13-Align-sequences-on-QIIME-using-the-ClustalW-or-muscle-method"><span class="toc-number">2.13.</span> <span class="toc-text">(13) Align sequences on QIIME using the ClustalW or muscle method</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#14-Make-the-reference-tree-on-QIIME"><span class="toc-number">2.14.</span> <span class="toc-text">(14) Make the reference tree on QIIME</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#15-%E5%A4%9A%E6%A0%B7%E6%80%A7%E5%88%86%E6%9E%90-Run-diversity-analyses-on-QIIME"><span class="toc-number">2.15.</span> <span class="toc-text">(15)多样性分析(Run diversity analyses on QIIME)</span></a></li></ol></li><li class="toc-item toc-level-3"><a class="toc-link" href="#3-Notes%EF%BC%9A"><span class="toc-number">3.</span> <span class="toc-text">3. Notes：</span></a></li></ol></div></div></div></div><div class="pure-u-1 pure-u-md-3-4"><div id="footer">Copyright © 2023 <a rel="nofollow" target="_blank" href="https://www.ligene.cn">LIGENE |</a> Powered by<a rel="nofollow" target="_blank" href="https://hexo.io"> Hexo |</a> Theme by<a rel="nofollow" target="_blank" href="https://github.com/tufu9441/maupassant-hexo"> Maupassant.</a></div></div></div><a class="show" id="rocket" href="#top"></a><script type="text/javascript" src="/js/totop.js?v=1.0.0" async></script><script type="text/javascript" src="//lib.baomitu.com/fancybox/latest/jquery.fancybox.min.js"></script><script type="text/javascript" src="/js/fancybox.js?v=1.0.0"></script><link rel="stylesheet" type="text/css" href="//lib.baomitu.com/fancybox/latest/jquery.fancybox.min.css"><link rel="stylesheet" type="text/css" href="/css/search.css?v=1.0.0"><script type="text/javascript" src="/js/search.js?v=1.0.0"></script><script>var search_path = 'search.xml';
if (search_path.length == 0) {
  search_path = 'search.xml';
}
var path = '/' + search_path;
searchFunc(path, 'local-search-input', 'local-search-result');
</script><script type="text/javascript" src="/js/copycode.js?v=1.0.0" successtext="复制成功！"></script><link rel="stylesheet" type="text/css" href="/css/copycode.css?v=1.0.0"><script type="text/javascript" src="/js/codeblock-resizer.js?v=1.0.0"></script><script type="text/javascript" src="/js/smartresize.js?v=1.0.0"></script></div></body></html>