<!DOCTYPE html><html lang="zh-CN"><head><meta http-equiv="content-type" content="text/html; charset=utf-8"><meta content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=0" name="viewport"><meta content="yes" name="apple-mobile-web-app-capable"><meta content="black-translucent" name="apple-mobile-web-app-status-bar-style"><meta content="telephone=no" name="format-detection"><meta name="description"><title>Mapping contigs onto reference genome | LIGENE</title><link rel="stylesheet" type="text/css" href="/css/style.css?v=1.0.0"><link rel="stylesheet" type="text/css" href="//lib.baomitu.com/normalize/latest/normalize.min.css"><link rel="stylesheet" type="text/css" href="//lib.baomitu.com/pure/latest/pure-min.min.css"><link rel="stylesheet" type="text/css" href="//lib.baomitu.com/pure/latest/grids-responsive-min.min.css"><link rel="stylesheet" href="//lib.baomitu.com/font-awesome/4.7.0/css/font-awesome.min.css"><script type="text/javascript" src="//lib.baomitu.com/jquery/latest/jquery.min.js"></script><link rel="icon" mask="" sizes="any" href="/favicon.ico"><link rel="Shortcut Icon" type="image/x-icon" href="/favicon.ico"><link rel="apple-touch-icon" href="/apple-touch-icon.png"><link rel="apple-touch-icon-precomposed" href="/apple-touch-icon.png"><link rel="alternate" type="application/atom+xml" href="/atom.xml"><script type="text/javascript" src="//lib.baomitu.com/clipboard.js/latest/clipboard.min.js"></script><script type="text/javascript" src="//lib.baomitu.com/toastr.js/latest/toastr.min.js"></script><link rel="stylesheet" href="//lib.baomitu.com/toastr.js/latest/toastr.min.css"><meta name="generator" content="Hexo 6.0.0"></head><script type="text/javascript" src="/js/toc.js?v=1.0.0"></script><body><div class="body_container"><div id="header"><div class="site-name"><h1 class="hidden">Mapping contigs onto reference genome</h1><a id="logo" href="/.">LIGENE</a><p class="description">Notes of Bioinformatics and Genomics</p></div><div id="nav-menu"><a class="current" href="/."><i class="fa fa-home"> 首页</i></a><a href="/archives/"><i class="fa fa-archive"> 归档</i></a><a href="/bigbook/"><i class="fa fa-book"> 教程</i></a><a href="/downloads/"><i class="fa fa-download"> 下载</i></a><a href="/history/"><i class="fa fa-tag"> 历史</i></a><a href="/about/"><i class="fa fa-user"> 关于</i></a></div></div><div class="pure-g" id="layout"><div class="pure-u-1 pure-u-md-3-4"><div class="content_container"><div class="post"><h1 class="post-title">Mapping contigs onto reference genome</h1><div class="post-meta">2017-07-13<span> | </span><span class="category"><a href="/categories/Genomics/">Genomics</a></span><span class="post-time"><span class="post-meta-item-text"> | </span><span class="post-meta-item-icon"><i class="fa fa-keyboard-o"></i><span class="post-count"> 439</span><span class="post-meta-item-text"> 字</span></span></span><span class="post-time"> | <span class="post-meta-item-icon"><i class="fa fa-clock-o"></i><span class="post-count"> 2</span><span class="post-meta-item-text"> 分钟</span></span></span></div><a class="disqus-comment-count" href="/2017/07/13/mapping-contigs/#vcomment"><span class="valine-comment-count" data-xid="/2017/07/13/mapping-contigs/"></span><span> 条评论</span></a><div class="post-content"><p>基因组组装会得到许多contigs，一般需要将它们比对到参考基因组序列，以确定contigs在染色体的位置。<br>参考基因组可以用亲缘关系很近的物种序列，例如同一个物种的不同菌株测序，可以用该种模式菌株的基因组序列作为reference sequence。</p>
<span id="more"></span>
<h2 id="1-Mapping-contigs-onto-reference-genome"><a href="#1-Mapping-contigs-onto-reference-genome" class="headerlink" title="1. Mapping contigs onto reference genome"></a>1. Mapping contigs onto reference genome</h2><h3 id="1-1-Make-the-BLAST-database-with-the-reference-genome"><a href="#1-1-Make-the-BLAST-database-with-the-reference-genome" class="headerlink" title="1.1 Make the BLAST database with the reference genome"></a>1.1 Make the BLAST database with the reference genome</h3><figure class="highlight plaintext"><table><tr><td class="code"><pre><span class="line">$makeblastdb -in reference.fasta -dbtype nucl -parse_seqids -out ref_db</span><br><span class="line">$blastdbcmd -db ref_db -info</span><br></pre></td></tr></table></figure>

<h3 id="1-2-Run-the-BLAST-query-with-contigs"><a href="#1-2-Run-the-BLAST-query-with-contigs" class="headerlink" title="1.2 Run the BLAST query with contigs"></a>1.2 Run the BLAST query with contigs</h3><p>$ blastn -task blastn -query contigs.fasta -db ref_db -parse_deflines  -evalue 0.0001 -outfmt 6 -out blast.tab </p>
<h3 id="1-3-Convert-BLAST-tabular-format-into-BED-format"><a href="#1-3-Convert-BLAST-tabular-format-into-BED-format" class="headerlink" title="1.3 Convert BLAST tabular format into BED format"></a>1.3 Convert BLAST tabular format into BED format</h3><p>BED format is a format that IGV can accept. You need to have a few column in this file format:</p>
<blockquote>
<p>“chrom start end name score, ….”<br>These fields are already present in the BLAST tabular format result:<br>‘qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore’<br>To convert to BED format, assuming query is your contigs and subject is the reference:<br>“subjectID, subjectStart - 1,  subjectEnd, queryID:queryStart-queryEnd”<br>Note: </p>
</blockquote>
<ol>
<li>you might need to swap subjectStart and subjectEnd, depending on which one is larger.</li>
<li>use tab instead of comma in the final BED file.</li>
</ol>
<h3 id="1-4-Visualization-with-IGV"><a href="#1-4-Visualization-with-IGV" class="headerlink" title="1.4 Visualization with IGV"></a>1.4 Visualization with IGV</h3><p>The final BED file can be visualized with IGV.</p>
<hr>
<h2 id="2-最新版BLAST-支持直接用IGV显示"><a href="#2-最新版BLAST-支持直接用IGV显示" class="headerlink" title="2. 最新版BLAST+支持直接用IGV显示"></a>2. 最新版BLAST+支持直接用IGV显示</h2><ul>
<li><p>最新版blastn支持SAM output format：**-outfmt 17**<br>参见：NCBI working on SAM output from BLAST+ (<a target="_blank" rel="noopener" href="https://blastedbio.blogspot.com/2015/07/ncbi-working-on-sam-output-from-blast.html">https://blastedbio.blogspot.com/2015/07/ncbi-working-on-sam-output-from-blast.html</a>)</p>
</li>
<li><p>但输出SAM格式还有一个问题，Query name不能被输出到sam文件对应的序列。<br>可以通过 -parse_deflines来解决：</p>
<figure class="highlight plaintext"><table><tr><td class="code"><pre><span class="line">$makeblastdb -dbtype nucl -in contigseq.fasta -parse_seqids -out testdb</span><br><span class="line">$blastn -query genomeseq.fasta -parse_deflines -db testdb -evalue 0.0001 -outfmt 17 -out test.sam</span><br></pre></td></tr></table></figure>
<p>注意：这里<strong>query是reference序列文件，而subject是contigs序列</strong>， 所以需要用contigs序列建库，与前面tabular format的过程相反。</p>
</li>
<li><p>输出文件test.sam可直接导入IGV中观察两基因组的比对情况。<br>  IGV需要以这里的genomeseq.fasta做为reference genome (Genomes-&gt;Create .genome file…).</p>
</li>
</ul>
</div><div class="tags"><a href="/tags/BLAST"><i class="fa fa-tag">BLAST</i></a><a href="/tags/IGV"><i class="fa fa-tag">IGV</i></a></div><div class="post-nav"><a class="pre" href="/2017/07/20/Microbiome-visualization/">Microbiome visualization</a><a class="next" href="/2017/07/05/BLAST-Tutorial/">BLAST+使用教程</a></div><div id="vcomment"></div><script src="//c.lcfile.com/static/js/av-core-mini-0.6.4.js"></script><script src="//unpkg.com/valine@latest/dist/Valine.min.js"></script><script>var notify = 'false' == 'true' ? true : false;
var verify = 'false' == 'true' ? true : false;
var GUEST_INFO = ['nick','mail','link'];
var guest_info = 'nick,mail'.split(',').filter(function(item){
  return GUEST_INFO.indexOf(item) > -1
});
guest_info = guest_info.length == 0 ? GUEST_INFO :guest_info;
window.valine = new Valine({
  el:'#vcomment',
  notify:notify,
  verify:verify,
  appId:'MNJKe5WJm5b2Ef1eGugrbjeW-9Nh9j0Va',
  appKey:'JidWqAC9XmARrOScdPmYsjOc',
  placeholder:'Writing comments here',
  avatar:'mm',
  guest_info:guest_info,
  pageSize:'10'
})
</script></div></div></div><div class="pure-u-1-4 hidden_mid_and_down"><div id="sidebar-toc"><div class="stoc-article" id="sidebar-stoc"><strong class="stoc-title"><i class="fa fa-bars"> Contents </i></strong><div class="toc-nav" id="stoc"><ol class="toc"><li class="toc-item toc-level-2"><a class="toc-link" href="#1-Mapping-contigs-onto-reference-genome"><span class="toc-number">1.</span> <span class="toc-text">1. Mapping contigs onto reference genome</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#1-1-Make-the-BLAST-database-with-the-reference-genome"><span class="toc-number">1.1.</span> <span class="toc-text">1.1 Make the BLAST database with the reference genome</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#1-2-Run-the-BLAST-query-with-contigs"><span class="toc-number">1.2.</span> <span class="toc-text">1.2 Run the BLAST query with contigs</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#1-3-Convert-BLAST-tabular-format-into-BED-format"><span class="toc-number">1.3.</span> <span class="toc-text">1.3 Convert BLAST tabular format into BED format</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#1-4-Visualization-with-IGV"><span class="toc-number">1.4.</span> <span class="toc-text">1.4 Visualization with IGV</span></a></li></ol></li><li class="toc-item toc-level-2"><a class="toc-link" href="#2-%E6%9C%80%E6%96%B0%E7%89%88BLAST-%E6%94%AF%E6%8C%81%E7%9B%B4%E6%8E%A5%E7%94%A8IGV%E6%98%BE%E7%A4%BA"><span class="toc-number">2.</span> <span class="toc-text">2. 最新版BLAST+支持直接用IGV显示</span></a></li></ol></div></div></div></div><div class="pure-u-1 pure-u-md-3-4"><div id="footer">Copyright © 2023 <a href="/." rel="nofollow">LIGENE.</a> Powered by<a rel="nofollow" target="_blank" href="https://hexo.io"> Hexo.</a><a rel="nofollow" target="_blank" href="https://github.com/tufu9441/maupassant-hexo"> Theme</a> by<a rel="nofollow" target="_blank" href="https://github.com/pagecho"> Cho.</a></div></div></div><a class="show" id="rocket" href="#top"></a><script type="text/javascript" src="/js/totop.js?v=1.0.0" async></script><script type="text/javascript" src="//lib.baomitu.com/fancybox/latest/jquery.fancybox.min.js"></script><script type="text/javascript" src="/js/fancybox.js?v=1.0.0"></script><link rel="stylesheet" type="text/css" href="//lib.baomitu.com/fancybox/latest/jquery.fancybox.min.css"><link rel="stylesheet" type="text/css" href="/css/search.css?v=1.0.0"><script type="text/javascript" src="/js/search.js?v=1.0.0"></script><script>var search_path = 'search.xml';
if (search_path.length == 0) {
  search_path = 'search.xml';
}
var path = '/' + search_path;
searchFunc(path, 'local-search-input', 'local-search-result');
</script><script type="text/javascript" src="/js/copycode.js?v=1.0.0" successtext="复制成功！"></script><link rel="stylesheet" type="text/css" href="/css/copycode.css?v=1.0.0"><script type="text/javascript" src="/js/codeblock-resizer.js?v=1.0.0"></script><script type="text/javascript" src="/js/smartresize.js?v=1.0.0"></script></div></body></html>