<!DOCTYPE html><html lang="zh-CN"><head><meta http-equiv="content-type" content="text/html; charset=utf-8"><meta content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=0" name="viewport"><meta content="yes" name="apple-mobile-web-app-capable"><meta content="black-translucent" name="apple-mobile-web-app-status-bar-style"><meta content="telephone=no" name="format-detection"><meta name="description" content="Notes of Bioinformatics &amp; Genomics"><title>数据分析的多重检验问题(multiple testing) | LIGENE</title><link rel="stylesheet" type="text/css" href="/css/style.css?v=1.0.0"><link rel="stylesheet" type="text/css" href="//cdn.jsdelivr.net/npm/normalize.css/normalize.min.css"><link rel="stylesheet" type="text/css" href="//cdn.jsdelivr.net/npm/purecss/build/pure-min.min.css"><link rel="stylesheet" type="text/css" href="//cdn.jsdelivr.net/npm/purecss/build/grids-responsive-min.css"><link rel="stylesheet" href="//cdn.jsdelivr.net/npm/font-awesome@4.7.0/css/font-awesome.min.css"><script type="text/javascript" src="//cdn.jsdelivr.net/npm/jquery/dist/jquery.min.js"></script><link rel="icon" mask="" sizes="any" href="/favicon.ico"><link rel="Shortcut Icon" type="image/x-icon" href="/favicon.ico"><link rel="apple-touch-icon" href="/apple-touch-icon.png"><link rel="apple-touch-icon-precomposed" href="/apple-touch-icon.png"><link rel="alternate" type="application/atom+xml" href="/atom.xml"><script type="text/javascript" src="//cdn.jsdelivr.net/npm/clipboard/dist/clipboard.min.js"></script><script type="text/javascript" src="//cdn.jsdelivr.net/gh/codeseven/toastr/build/toastr.min.js"></script><link rel="stylesheet" href="//cdn.jsdelivr.net/gh/codeseven/toastr/build/toastr.min.css"><meta name="generator" content="Hexo 6.0.0"></head><script type="text/javascript" src="/js/toc.js?v=1.0.0"></script><body><div class="body_container"><div id="header"><div class="site-name"><h1 class="hidden">数据分析的多重检验问题(multiple testing)</h1><a id="logo" href="/.">LIGENE</a><p class="description">Notes of Bioinformatics and Genomics</p></div><div id="nav-menu"><a class="current" href="/."><i class="fa fa-home"> 首页</i></a><a href="/archives/"><i class="fa fa-archive"> 归档</i></a><a href="/bigbook/"><i class="fa fa-book"> 教程</i></a><a href="/downloads/"><i class="fa fa-download"> 下载</i></a><a href="/history/"><i class="fa fa-tag"> 历史</i></a><a href="/about/"><i class="fa fa-user"> 关于</i></a></div></div><div class="pure-g" id="layout"><div class="pure-u-1 pure-u-md-3-4"><div class="content_container"><div class="post"><h1 class="post-title">数据分析的多重检验问题(multiple testing)</h1><div class="post-meta">2017-08-26<span> | </span><span class="category"><a href="/categories/Statistics/">Statistics</a></span></div><a class="disqus-comment-count" href="/2017/08/26/multiple-testing/#vcomment"><span class="valine-comment-count" data-xid="/2017/08/26/multiple-testing/"></span><span> 条评论</span></a><div class="post-content"><p>数据分析中常碰见多重检验(multiple testing)问题，如利用基因芯片数据比较分析两个样品间是否有差异表达基因，或利用质谱定量数据鉴定某一个蛋白在两个样品中是否存在差异表达等。<br>一般选取差异表达基因有两标准：<br>(1) Fold Change，即两个样品中同一个基因的表达水平变化倍数;<br>(2) p-value，即通过T检验等统计学方法计算每个基因的P值。</p>
<span id="more"></span>
<p>但是由于通常样本量较少，从而对总体方差的估计不很准确，所以T检验的检验效能会降低，并且如果多次使用T检验会显著增加假阳性的次数。例如，当某个基因的p值小于0.05（5%）时，我们通常认为这个基因在两个样本中的表达是有差异的。但是仍旧有5%的概率，这个基因并不是差异表达基因，我们错误地否认了原假设（在两个样本中没有差异表达），导致了假阳性的产生（犯错的概率为5%）。<br>如果检验一次，犯错的概率是5%；检测10000次，犯错的次数就是500次，即额外多出了500个差异基因的结论（即使实际没有差异）。为了控制假阳性的次数，于是我们需要对p值进行多重检验校正，提高阈值。</p>
<h3 id="FDR方法"><a href="#FDR方法" class="headerlink" title="FDR方法"></a>FDR方法</h3><p>Benjamini于1995年提出一种方法，通过控制FDR(False Discovery Rate)来决定P值的域值。假设你挑选了R个差异表达的基因，其中有S个是真正有差异表达的，另外有V个其实是没有差异表达的，是假阳性的。实践中希望错误比例Q＝V&#x2F;R平均而言不能超过某个预先设定的值（比如0.05），在统计学上，这也就等价于控制FDR不能超过5％.<br>根据Benjamini在他的文章中所证明的定理，控制FDR的步骤实际上非常简单。<br>设总共有m个候选基因，每个基因对应的p值从小到大排列分别是p(1),p(2),…,p(m),则若想控制FDR不能超过q，则只需找到最大的正整数i，使得 p(i)&lt;&#x3D; (i*q)&#x2F;m。然后，挑选对应p(1),p(2),…,p(i)的基因做为差异表达基因，这样就能从统计学上保证FDR不超过q。这是Benjamini and Hochber法(BH法)从P值估计FDR的基本原理。</p>
<h3 id="FDR计算"><a href="#FDR计算" class="headerlink" title="FDR计算"></a>FDR计算</h3><p>FDR的计算方法除了可以使用excel的BH计算方法外，对于较大的数据，我们推荐使用R命令p.adjust。<br><img src="https://raw.githubusercontent.com/adong77/LIGENE/master/imageBed/fdr_padjust.png" alt="p.adjust"><br>该函数的运算需要先将一系列p值、设定一个校正方法（p.adjust.methods）及所有p值的个数（length(p)）输入到p.adjust函数中。返回结果为校正后的P值，可以用于后续分析了。</p>
<h3 id="FDR与p-value区别"><a href="#FDR与p-value区别" class="headerlink" title="FDR与p-value区别"></a>FDR与p-value区别</h3><p>The False Discovery Rate (FDR) of a set of predictions is the expected percent of false predictions in the set of predictions . For example if the algorithm returns 100 genes with a false discovery rate of .3 then we should expect 70 of them to be correct. FDR的意义为错误拒绝（即拒绝真的原假设）的个数占所有被拒绝的原假设个数的比例的期望值！<br>The FDR is very different from a p-value, and as such a much higher FDR can be tolerated than with a p-value. In the example above a set of 100 predictions of which 70 are correct might be very useful, especially if there are thousands of genes on the array most of which are not differentially expressed. In contrast p-value of .3 is generally unacceptabe in any circumstance. Meanwhile an FDR of as high as .5 or even higher might be quite meaningful.</p>
</div><div class="tags"><a href="/tags/FDR/"><i class="fa fa-tag"></i>FDR</a></div><div class="post-nav"><a class="pre" href="/2017/09/03/qiime-install/">Ubuntu 16.04LTS安装QIIME</a><a class="next" href="/2017/08/26/binfo-JS/">Javascript-based bioinformatics</a></div><div id="vcomment"></div><script src="//cdn1.lncld.net/static/js/3.0.4/av-min.js"></script><script src="//unpkg.com/valine@latest/dist/Valine.min.js"></script><script>var notify = 'false' == 'true' ? true : false;
var verify = 'false' == 'true' ? true : false;
var GUEST_INFO = ['nick','mail','link'];
var guest_info = 'nick,mail,link'.split(',').filter(function(item){
  return GUEST_INFO.indexOf(item) > -1
});
guest_info = guest_info.length == 0 ? GUEST_INFO :guest_info;
window.valine = new Valine({
  el:'#vcomment',
  notify:notify,
  verify:verify,
  appId:'MNJKe5WJm5b2Ef1eGugrbjeW-9Nh9j0Va',
  appKey:'JidWqAC9XmARrOScdPmYsjOc',
  placeholder:'Writing comments here',
  avatar:'mm',
  guest_info:guest_info,
  pageSize:'10'
})
</script></div></div></div><div class="pure-u-1-4 hidden_mid_and_down"><div id="sidebar-toc"><div class="stoc-article" id="sidebar-stoc"><strong class="stoc-title"><i class="fa fa-blind"> Contents </i></strong><div class="toc-nav" id="stoc"><ol class="toc"><li class="toc-item toc-level-3"><a class="toc-link" href="#FDR%E6%96%B9%E6%B3%95"><span class="toc-text">FDR方法</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#FDR%E8%AE%A1%E7%AE%97"><span class="toc-text">FDR计算</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#FDR%E4%B8%8Ep-value%E5%8C%BA%E5%88%AB"><span class="toc-text">FDR与p-value区别</span></a></li></ol></div></div></div></div><div class="pure-u-1 pure-u-md-3-4"><div id="footer">Copyright © 2023 <a target="_blank" href="http://www.ligene.cn" rel="nofollow noopener">LIGENE.</a> Powered by<a rel="nofollow" target="_blank" href="https://hexo.io"> Hexo</a> & GitHub Pages.</div></div></div><a class="show" id="rocket" href="#top"></a><script type="text/javascript" src="/js/totop.js?v=1.0.0" async></script><script type="text/javascript" src="//cdn.jsdelivr.net/gh/fancyapps/fancybox/dist/jquery.fancybox.min.js" async></script><script type="text/javascript" src="/js/fancybox.js?v=1.0.0" async></script><link rel="stylesheet" type="text/css" href="//cdn.jsdelivr.net/gh/fancyapps/fancybox/dist/jquery.fancybox.min.css"><link rel="stylesheet" type="text/css" href="/css/search.css?v=1.0.0"><script type="text/javascript" src="/js/search.js?v=1.0.0"></script><script>var search_path = 'search.xml';
if (search_path.length == 0) {
   search_path = 'search.xml';
}
var path = '/' + search_path;
searchFunc(path, 'local-search-input', 'local-search-result');
</script><script type="text/javascript" src="/js/copycode.js" successtext="复制成功!"></script><link rel="stylesheet" type="text/css" href="/css/copycode.css"><script type="text/javascript" src="/js/codeblock-resizer.js?v=1.0.0"></script><script type="text/javascript" src="/js/smartresize.js?v=1.0.0"></script></div></body></html>